### PR TITLE
filterx: refactors/fixes and convert funcs to generators

### PR DESCRIPTION
**Pull request by MrAnno**

### CODE REVIEW

File: lib/filterx/expr-literal-generator.c [hunk 1]
```diff
@@ -150,6 +150,22 @@ filterx_literal_dict_generator_foreach(FilterXExpr *s, FilterXLiteralDictGenerat
...
+gboolean
+filterx_literal_list_generator_foreach(FilterXExpr *s, FilterXLiteralListGeneratorForeachFunc func, gpointer user_data)
+{
+  FilterXExprLiteralGenerator *self = (FilterXExprLiteralGenerator *) s;
+
+  for (GList *link = self->elements; link; link = link->next)
+    {
+      FilterXLiteralGeneratorElem *elem = (FilterXLiteralGeneratorElem *) link->data;
+
+      if (!func(elem->value, user_data))
```
Comment (OverOrion) [inline lib/filterx/expr-literal-generator.c:162]: Can we also expose the `index` of the element as well to the user supplied function?

Comment (MrAnno) [reply lib/filterx/expr-literal-generator.c:162]: Good idea.

**Review by alltilla — APPROVED** [review]

**Comment by alltilla** [comment]
Approve, keeping the PR open so you can include index in the list foreach.

**Review by alltilla — DISMISSED** [review]

**Comment by alltilla** [comment]
Sorry, my PR introduced a merge conflict here. Do you want me to fix it?

**Comment by MrAnno** [comment]
I didn't notice. Rebased :)

**Review by alltilla — APPROVED** [review]

**Pull request merged**